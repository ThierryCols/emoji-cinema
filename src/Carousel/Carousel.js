import React from 'react';

const Carousel = (props) => (
  <div className="carousel">
    {props.movieList.filter(
      (movie) => movie.title.toLowerCase().includes(props.query.toLowerCase())
    ).map((movie) => (
      <div className="movie" key={movie.title}>
        <p className="title">{movie.title} - </p>
        <p className="emoji">{movie.emoji}</p>
      </div>
    ))}
  </div>
)

export default Carousel;
