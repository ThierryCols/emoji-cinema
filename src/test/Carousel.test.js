import React from 'react';
import Carousel from '../Carousel/Carousel';
import renderer from 'react-test-renderer';
import movieList from '../store'

it('renders correctly', () => {
  const tree = renderer.create(
    <Carousel
      movieList={movieList}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
