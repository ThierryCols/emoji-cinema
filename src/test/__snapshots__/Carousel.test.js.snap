// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="carousel"
/>
`;

exports[`renders correctly 2`] = `
<div
  className="carousel"
/>
`;

exports[`renders correctly 3`] = `
<div
  className="carousel"
>
  <div
    className="movie"
  >
    <p
      className="title"
    >
      Frozen 2
       - 
    </p>
    <p
      className="emoji"
    >
      ❄️👗✌️
    </p>
  </div>
</div>
`;

exports[`renders correctly 4`] = `
<div
  className="carousel"
/>
`;
